src/vcf/odb_report.cpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "vcf/odb_report.hpp"
+
+namespace ebi
+{
+  namespace vcf
+  {
+    OdbReportRW::OdbReportRW(const std::string &db_name) : db_name(db_name), current_transaction_size{0},
+                                                          transaction_size{1000000}
+    {
+        try {
+            db = std::unique_ptr<odb::sqlite::database> (
+                    new odb::sqlite::database{
+                            db_name, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE});
+
+            // Create the database schema. Due to bugs in SQLite foreign key
+            // support for DDL statements, we need to temporarily disable
+            // foreign keys.
+            //
+            {
+                odb::core::connection_ptr c{db->connection()};
+
+                c->execute("PRAGMA foreign_keys=OFF");
+
+                odb::core::transaction t{c->begin()};
+                odb::core::schema_catalog::create_schema(*db);
+                t.commit();
+
+                c->execute("PRAGMA foreign_keys=ON");
+            }
+        } catch (const odb::exception& e) {
+            throw std::runtime_error{std::string{"ODB report: failed ODB initialization: "} + e.what()};
```

Comment:
First part of the message could be a bit more user friendly, similar to sqlite_report.cpp: `"Can't initialize database: " + e.what()`.